flamily/irs
# Add script for automated setup, pre-commit hook updates and organize doccos

**This PR does the following:**
- automated environment setup which will create the db for you, load the schema, set the path for pipenv installation, install pipenv and pre-commit git hooks
- I added pylint in addition to flake8 for the actual `pre-commit` git command, so when you run that, it will do the original pre-commit checks plus pylint check (with the exception of docstring) since I saw you recently add pylint as part of the code quality check and we might as well do it automatically.
- symlink pre-commit git hook so it actually does all those checks _when the users do the commit_ (meaning you no longer have to do run `pre-commit` at all to do the checks, it automatically does it for you. Win!
- updated the documentation. It's starting to become long and cumbersome. I simply moved it out so and referenced the specific document parts when necessary just so the peoples will only see what's immediately relevant to them.

_Note: you can still run pylint and pre-commit manually but this just means people can't "forget" to run it_ ;)

## Results (if you're wondering)
### Automated setup script output when i was testing:
![screenshot from 2018-10-05 07-52-50](https://user-images.githubusercontent.com/20508769/46507264-067a2180-c87b-11e8-8897-18d9f3653158.png)

### pre-commit changes
![screenshot from 2018-10-05 07-02-38](https://user-images.githubusercontent.com/20508769/46507295-290c3a80-c87b-11e8-9856-30874b8b2d2d.png)

## To test: 
Automated setup - Run the setup script. Report any errors to me.
Pre-commit hook - Screw up your files then stage your changes and do a commit. See that it does the flake8, pylint etc. checks automatically.
